### PR TITLE
Upgrade to on-ramp-sdk@1.3.1: Wyre Apple Pay auth support and inAppBrowser hook

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR upgrades the `on-ramp-sdk` to version `1.3.1`.
It adds support for Wyre Apple Pay `authorizationUrl` which is included in this SDK version. 
A new hook for the inAppBrowser is added.

**Screenshots/Recordings**

~~_If applicable, add screenshots and/or recordings to visualize the before and after of your change_~~

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/containers/ApplePayButton.tsx [hunk 1]
```diff
@@ -99,20 +173,26 @@ const ApplePayButton = ({
...
-    selectedAddress,
-    trackEvent,
-    dispatch,
   ]);
 
-  return <ApplePayButtonComponent label={label} onPress={handlePress} />;
+  return (
+    <>
+      {authorizationResult?.authenticationUrl ? (
+        <WebView
```
Comment (Cal-L) [inline app/components/UI/FiatOnRampAggregator/containers/ApplePayButton.tsx:188]: Let's comment that this is a invisible/placeholder webview

File: app/components/UI/FiatOnRampAggregator/containers/CustomActionButton.tsx [hunk 2]
```diff
@@ -85,138 +66,61 @@ const CustomActionButton: React.FC<
         payment_method_id: selectedPaymentMethodId as string,
       });
 
-      const redirectUrl = `${callbackBaseDeeplink}on-ramp${providerId}`;
-
-      const { url, orderId: customOrderId } = await sdk.getBuyUrl(
+      const buyAction = await sdk.getBuyUrl(
```
Comment (Cal-L) [inline app/components/UI/FiatOnRampAggregator/containers/CustomActionButton.tsx:69]: In the future, would it make sense to combine this and the logic in GetQuotes to be a hook?

File: app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      buyAction: BuyAction,
+      provider: Provider,
+      amount?: number,
+      fiatSymbol?: string,
+    ) => {
+      const deeplinkRedirectUrl = `${callbackBaseDeeplink}on-ramp${provider.id}`;
+      const { url, orderId: customOrderId } = await buyAction.createWidget(
+        deeplinkRedirectUrl,
+      );
+
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts:97]: check of `customOrderId` not being `undefined` before calling `createCustomOrderIdData`

**Review by Cal-L — APPROVED** [review]
LGTM

File: app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts (hunk 3 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/hooks/useInAppBrowser.ts:97]: Addressed here https://github.com/MetaMask/metamask-mobile/pull/5269

File: app/components/UI/FiatOnRampAggregator/containers/CustomActionButton.tsx (hunk 2 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/containers/CustomActionButton.tsx:69]: Yes, the same as `handleSuccessfulOrder`, these are potentially new hooks

File: app/components/UI/FiatOnRampAggregator/containers/ApplePayButton.tsx (hunk 1 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/containers/ApplePayButton.tsx:188]: Added in https://github.com/MetaMask/metamask-mobile/pull/5269

**Pull request merged**